From 9b3d307f8f6a1af88f3f810f5a6cf0835830e1e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Krause?= <joerg.krause@embedded.rocks>
Date: Mon, 7 Dec 2015 01:14:33 +0100
Subject: [PATCH] add executable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a new --executable option to distribute so that we can
force the shebang line in installed python scripts.

[Thomas: refresh for setuptools 5.8.]
[Jörg: refresh for setuptools 18.7.1]

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Jörg Krause <joerg.krause@embedded.rocks>
---
 setuptools/command/install.py         | 2 ++
 setuptools/command/install_scripts.py | 9 +++++++++
 2 files changed, 11 insertions(+)

diff --git a/setuptools/command/install.py b/setuptools/command/install.py
index 72b9a3e..6781d2b 100644
--- a/setuptools/command/install.py
+++ b/setuptools/command/install.py
@@ -16,6 +16,7 @@ class install(orig.install):
     """Use easy_install to install the package, w/dependencies"""
 
     user_options = orig.install.user_options + [
+        ('executable=', 'e', "specify final destination interpreter path"),
         ('old-and-unmanageable', None, "Try not to use this!"),
         ('single-version-externally-managed', None,
          "used by system package builders to create 'flat' eggs"),
@@ -38,6 +39,7 @@ class install(orig.install):
         )
 
         orig.install.initialize_options(self)
+        self.executable = None
         self.old_and_unmanageable = None
         self.single_version_externally_managed = None
 
diff --git a/setuptools/command/install_scripts.py b/setuptools/command/install_scripts.py
index 9cd8eb0..7786150 100644
--- a/setuptools/command/install_scripts.py
+++ b/setuptools/command/install_scripts.py
@@ -13,6 +13,13 @@ class install_scripts(orig.install_scripts):
     def initialize_options(self):
         orig.install_scripts.initialize_options(self)
         self.no_ep = False
+        self.executable = None
+
+    def finalize_options(self):
+        orig.install_scripts.finalize_options(self)
+        self.set_undefined_options('install',
+                ('executable','executable')
+        )
 
     def run(self):
         import setuptools.command.easy_install as ei
@@ -33,6 +40,8 @@ class install_scripts(orig.install_scripts):
         )
         bs_cmd = self.get_finalized_command('build_scripts')
         exec_param = getattr(bs_cmd, 'executable', None)
+        if self.executable is not None:
+            exec_param = self.executable
         try:
             bw_cmd = self.get_finalized_command("bdist_wininst")
             is_wininst = getattr(bw_cmd, '_is_running', False)
-- 
2.30.2

